### ISSUE TITLE
Java TLS issues with Spring Cloud Vault and Vault 1.40

### ISSUE THREAD

**Issue by fernando-villalba** [hashicorp/vault#8750]
We are experiencing these issues in the latest version of vault (1.4.0):

https://github.com/golang/go/issues/35722

https://github.com/golang/go/issues/37808

This is the error we are getting from the microservice trying to connect to Vault:

`6-04-2020 10:48:57.821 [tracer: ]  WARN 1 --- [           main] LeaseEventPublisher$LoggingErrorListener : [RequestedSecret [path='kv/path/secret/blah', mode=ROTATE]] Lease [leaseId='null', leaseDuration=PT0S, renewable=false] Cannot login using org.springframework.web.client.ResourceAccessException: I/O error on POST request for "https://vault.address.com:443/v1/auth/aws/login": extension (5) should not be presented in certificate_request; nested exception is javax.net.ssl.SSLHandshakeException: extension (5) should not be presented in certificate_request`

This ends up killing the Microservice.

For now I reverted to version 1.3.4 and all working.

**Comment by pavankjadda** [comment]
I encountered this recently. Switched to TLS 1.2 using command line arguments for the time being. You can find the workaround [here](https://github.com/golang/go/issues/35722#issuecomment-571173416)

**Comment by fernando-villalba** [comment]
Updating to the latest version of Java 11 also worked for us...

**Comment by pavankjadda** [comment]
@fernando-villalba  I am using java 11.0.7 version and I still see this 

Java Version:
```
java 11.0.7 2020-04-14 LTS
Java(TM) SE Runtime Environment 18.9 (build 11.0.7+8-LTS)
Java HotSpot(TM) 64-Bit Server VM 18.9 (build 11.0.7+8-LTS, mixed mode)
```

**Comment by raj-saxena** [comment]
@pavankjadda I can confirm that upgrading to the latest OpenJDK version 11.0.7 resolves the issue.
```
$ java -version
openjdk version "11.0.7" 2020-04-14
OpenJDK Runtime Environment AdoptOpenJDK (build 11.0.7+10)
OpenJDK 64-Bit Server VM AdoptOpenJDK (build 11.0.7+10, mixed mode)
```

**Comment by cipherboy** [comment]
Closing this as a duplicate of https://github.com/golang/go/issues/35722 

As mentioned above, this was non-conformant behavior in the JDK's TLSv1.3 implementation, fixed in https://bugs.openjdk.java.net/browse/JDK-8236039 

The ability to allow JDK clients to provide OCSP information on connections is unresolved but here: https://bugs.openjdk.java.net/browse/JDK-8237486 -- but to my knowledge, this should not prevent connections. 

TL;DR: update JDK versions :-)
